autoload/neomake/utils.vim

```diff
@@ -200,3 +200,20 @@ function! neomake#utils#GetHighlight(group, what) abort
   endif
   return val
 endfunction
+
+function! neomake#utils#ExpandArgs(args) abort
+    if neomake#utils#IsRunningWindows()
+        call neomake#utils#ExpandArgsInWindows(a:args)
+    else
+        call neomake#utils#ExpandArgsInLinux(a:args)
+    endif
+endfunction
+
+function! neomake#utils#ExpandArgsInWindows(args) abort
+    " Don't expand &shellcmdflag argument of cmd.exe nor arguments like '"something"'
+    call map(a:args, "(v:val ==? &shellcmdflag || v:val =~? '.*\".*\".*') ? v:val : expand(v:val)")
+endfunction
+
+function! neomake#utils#ExpandArgsInLinux(args) abort
+    call map(a:args, 'expand(v:val)')
```

Comment:
I've tried setting up a docker instance of Arch but couldn't make it work (pacman seems not working), will look into this later.

OTOH, if I did the math right, we have the following makers relying on this feature:

```
ft/cf3.vim:10:        \ 'args': ['-cf', '%:p'],
ft/nix.vim:10:        \ 'args': ['%:p', '--parse-only'],
ft/elixir.vim:18:      \ 'args': ['credo', 'list', '%:p', '--format=oneline',
ft/elixir.vim:27:      \ 'args': ['dogma', '%:p', '--format=flycheck'],
ft/moon.vim:7:        \ 'args': ['-l', '%:p'],
ft/php.vim:39:        \ 'args': ['%:p', 'text', 'codesize,design,unusedcode,naming'],
ft/c.vim:26:        \ 'args': ['%:p', '--', '-Wall', '-Wextra'],
ft/typescript.vim:23-            \ '%:p', '--format verbose'
```

As you can see, they take advantage of this in order to handle the position of the file path `%:p`

Taking your proposal into account, we might take one of the following approaches:

- expand only when the arg starts with a Vim expansion char: `%` `#` `<`
- do not expand and let the makers' authors handle the expansions, ie: `[expand('%:p'), 'another-arg']`

And yes, whichever approach we pick, it might save us the expansion issues we have in Windows.